src/basilisp/core.lpy

```diff
@@ -3416,7 +3416,8 @@
 (defn rand-nth
   "Return a random element from ``coll``\\."
   [coll]
-  (nth coll (rand-int (count coll))))
+  (when coll
+    (nth coll (rand-int (count coll)))))
```

Comment:
The issue with this isn't the `nil`-ness of `coll`, but rather the error arising from `(rand-int 0)`.

More probably we need to do something like this:

```clojure
(defn rand-int
  "Return a random integer between ``lower`` (default: 0) and ``upper`` inclusive."
  ([upper]
   (rand-int 0 upper))
  ([lower upper]
   (cond
      (= upper lower) upper
      (< upper lower) (random/randrange upper lower)
      :else           (random/randrange lower upper)))
```

I think no changes will be required to `rand-nth` with this modification to `rand-int`.